src/conv/problem_description.cpp

```diff
@@ -96,6 +107,8 @@ void ProblemDescription::Serialize(std::ostream& stream) const
     stream << sep << PrintDHW('x', GetSpatialDims(), GetDilationD(), GetDilationH(), GetDilationW());
     stream << sep << GetBias();
     stream << sep << GetInLayout();
+    stream << sep << GetWeightsLayout();
```

Comment:
We must maintain this function backward compatible as it generates database keys. When layouts are default (NCHW-NKHW-KCYX) the modified function must return literally the same key as it does now.

/cc @JehandadKhan as this requirement is also important for Tuna, I think.

Reply:
Thanks. Will branch according to whether it is default layouts.